packages/blockchain/src/index.ts

```diff
@@ -128,14 +136,30 @@ export default class Blockchain implements BlockchainInterface {
       })
     }
 
-    this._validatePow = opts.validatePow !== undefined ? opts.validatePow : true
+    this._validateConsensus = opts.validateConsensus !== undefined ? opts.validateConsensus : true
     this._validateBlocks = opts.validateBlocks !== undefined ? opts.validateBlocks : true
 
     this.db = opts.db ? opts.db : level()
     this.dbManager = new DBManager(this.db, this._common)
 
-    if (this._validatePow) {
-      this.ethash = new Ethash(this.db)
+    if (this._validateConsensus) {
+      if (this._common.consensusType() !== 'pow') {
+        throw new Error('consensus validation only supported for pow chains')
+      }
+      if (this._common.consensusAlgorithm() !== 'ethash') {
+        throw new Error('consensus validation only supported for pow ethash algorithm')
+      }
+
+      this._ethash = new Ethash(this.db)
```

Comment:
Maybe it would be reasonable to allow users to provide ethash instance with options? It's using in examples to let user optimizae blockchain work, so it seems rational to accept users instance, rather than let get into the guts?

Reply:
I think this is something for a dedicated issue. I would put this under the kinds of feature requests though which we should only tackle if there is a demand expressed, otherwise we run into the risk of implementing things just on guess which at the end are not really used.